string label_name[]

---
bool flag